src/test/java/org/folio/dao/DataImportQueueDaoTest.java

```diff
@@ -54,118 +66,170 @@ public void setUp() {
+        any(Tuple.class),
+        ArgumentMatchers.<Promise<RowSet<Row>>>any()
+      );
 
     // when
     DataImportQueueItem queueItem = new DataImportQueueItem();
     queueItem.setFilePath("test/file/path");
     storedItemUUID = UUID.randomUUID();
     queueItem.setId(storedItemUUID.toString());
-    queueItem.setUploadDefinitionId(UUID.randomUUID()
-      .toString());
-    queueItem.setJobExecutionId(UUID.randomUUID()
-      .toString());
+    queueItem.setUploadDefinitionId(UUID.randomUUID().toString());
+    queueItem.setJobExecutionId(UUID.randomUUID().toString());
     queueItem.setSize(1000);
     queueItem.setOriginalSize(5000);
     DateTime now = new DateTime();
     queueItem.setTimestamp(now.toString());
-    queueItemDaoImpl.addQueueItem(queueItem)
+    queueItemDaoImpl
+      .addQueueItem(queueItem)
       // then
-      .onComplete(context.asyncAssertSuccess(x -> {
-        verify(postgresClient, times(1)).execute(anyString(), any(Tuple.class), ArgumentMatchers.<Promise<RowSet<Row>>>any());
-        verifyNoMoreInteractions(postgresClient);
-      }));
+      .onComplete(
+        context.asyncAssertSuccess(x -> {
+          verify(postgresClient, times(1))
+            .execute(
+              startsWith("INSERT INTO data_import_global.queue_items "),
+              any(Tuple.class),
+              ArgumentMatchers.<Promise<RowSet<Row>>>any()
+            );
+          verifyNoMoreInteractions(postgresClient);
+        })
+      );
   }
 
   @Test
-  public void shouldGetQueueItemById(TestContext context) {
+  public void shouldGetQueueItemByIdFailure(TestContext context) {
+    Async async = context.async();
+
     // given
     doAnswer((InvocationOnMock invocation) -> {
-      Promise<RowSet<Row>> promise = invocation.getArgument(2);
-      promise.complete(new LocalRowSet(3));
-      return null;
-    }).when(postgresClient)
-      .select(anyString(), any(Tuple.class), ArgumentMatchers.<Promise<RowSet<Row>>>any());
+        Promise<RowSet<Row>> promise = invocation.getArgument(2);
+        promise.complete(new LocalRowSet(3));
+        return null;
+      })
+      .when(postgresClient)
+      .select(
+        anyString(),
+        any(Tuple.class),
+        ArgumentMatchers.<Promise<RowSet<Row>>>any()
+      );
 
     // when
-    queueItemDaoImpl.getQueueItemById("sdfasdfasdf")
+    queueItemDaoImpl
+      .getQueueItemById("sample-id")
       // then
-      .onComplete(context.asyncAssertSuccess(x -> {
-        verify(postgresClient, times(1)).select(anyString(), any(Tuple.class), ArgumentMatchers.<Promise<RowSet<Row>>>any());
+      .onFailure(err -> {
+        verify(postgresClient, times(1))
+          .select(
+            anyString(),
+            any(Tuple.class),
+            ArgumentMatchers.<Promise<RowSet<Row>>>any()
+          );
         verifyNoMoreInteractions(postgresClient);
-      }));
+
+        assertThat(err, is(instanceOf(NoSuchElementException.class)));
+
+        async.complete();
```

Comment:
Since we're not getting an `AsyncResult`, the `context.assertAsyncSuccess` will not work.  However, we can manually control success/failure by running `context.async()` at the top of the test then running `async.success()` and `async.fail()` when the test succeeds/fails